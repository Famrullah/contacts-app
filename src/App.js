import React, {useEffect} from 'react';
import { fetchContacts } from "./store/actions/index";
import { useDispatch,useSelector } from 'react-redux'

function App() {
  const dispatch = useDispatch()
  
  const {data,isLoadingData} = useSelector(
    (state) => state
  );

  useEffect(() => {
    dispatch(fetchContacts())
  },[dispatch]);

  return (
    <div className="App">
      {isLoadingData?<div>loading</div>:<div>ini contoh aja</div>}
    </div>
  );
}

export default App;
