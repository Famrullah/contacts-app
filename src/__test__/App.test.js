import React from 'react';
import { render } from '@testing-library/react';
import App from '../App';
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'

describe('With React Testing Library', () => {
  const initialState = {output:10}
  const mockStore = configureStore()
  let store

  it('should render app and snapshot"', () => {
    store = mockStore(initialState)
    const { asFragment } = render(<Provider store={store}><App /></Provider>)
    expect(asFragment(<App />)).toMatchSnapshot();
  })
})