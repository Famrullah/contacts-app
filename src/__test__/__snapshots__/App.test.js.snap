// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With React Testing Library should render app and snapshot" 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="navbar"
    >
      <div
        class="contacts-app-container"
      >
        ini navbar
      </div>
    </div>
    <div
      class="contacts-app-container"
    >
      <div
        class="contact"
      >
        <div
          class="contact-list"
        >
          <div
            class="contact-list__desc"
          >
            <img
              alt=""
              class="profile-photo circle"
              src="https://source.unsplash.com/random/1"
            />
            <div
              class="offset-left-20"
            >
              <h3>
                 Faisal Amrullah
              </h3>
              <span>
                amrullahfaisal@gmail.com
              </span>
              <br />
              <span>
                081806868501
              </span>
            </div>
          </div>
          <div
            class="contact-list__action"
          >
            <span>
              Count : 0
            </span>
            <button
              class="btn"
            >
              Edit
            </button>
          </div>
        </div>
        <div
          class="contact-list"
        >
          <div
            class="contact-list__desc"
          >
            <img
              alt=""
              class="profile-photo circle"
              src="https://source.unsplash.com/random/2"
            />
            <div
              class="offset-left-20"
            >
              <h3>
                 Faisal Amrullah
              </h3>
              <span>
                amrullahfaisal@gmail.com
              </span>
              <br />
              <span>
                081806868501
              </span>
            </div>
          </div>
          <div
            class="contact-list__action"
          >
            <span>
              Count : 0
            </span>
            <button
              class="btn"
            >
              Edit
            </button>
          </div>
        </div>
        <div
          class="contact-list"
        >
          <div
            class="contact-list__desc"
          >
            <img
              alt=""
              class="profile-photo circle"
              src="https://source.unsplash.com/random/3"
            />
            <div
              class="offset-left-20"
            >
              <h3>
                 Faisal Amrullah
              </h3>
              <span>
                amrullahfaisal@gmail.com
              </span>
              <br />
              <span>
                081806868501
              </span>
            </div>
          </div>
          <div
            class="contact-list__action"
          >
            <span>
              Count : 0
            </span>
            <button
              class="btn"
            >
              Edit
            </button>
          </div>
        </div>
        <div
          class="contact-list"
        >
          <div
            class="contact-list__desc"
          >
            <img
              alt=""
              class="profile-photo circle"
              src="https://source.unsplash.com/random/4"
            />
            <div
              class="offset-left-20"
            >
              <h3>
                 Faisal Amrullah
              </h3>
              <span>
                amrullahfaisal@gmail.com
              </span>
              <br />
              <span>
                081806868501
              </span>
            </div>
          </div>
          <div
            class="contact-list__action"
          >
            <span>
              Count : 0
            </span>
            <button
              class="btn"
            >
              Edit
            </button>
          </div>
        </div>
        <div
          class="contact-list"
        >
          <div
            class="contact-list__desc"
          >
            <img
              alt=""
              class="profile-photo circle"
              src="https://source.unsplash.com/random/5"
            />
            <div
              class="offset-left-20"
            >
              <h3>
                 Faisal Amrullah
              </h3>
              <span>
                amrullahfaisal@gmail.com
              </span>
              <br />
              <span>
                081806868501
              </span>
            </div>
          </div>
          <div
            class="contact-list__action"
          >
            <span>
              Count : 0
            </span>
            <button
              class="btn"
            >
              Edit
            </button>
          </div>
        </div>
        <div
          class="contact-list"
        >
          <div
            class="contact-list__desc"
          >
            <img
              alt=""
              class="profile-photo circle"
              src="https://source.unsplash.com/random/6"
            />
            <div
              class="offset-left-20"
            >
              <h3>
                 Faisal Amrullah
              </h3>
              <span>
                amrullahfaisal@gmail.com
              </span>
              <br />
              <span>
                081806868501
              </span>
            </div>
          </div>
          <div
            class="contact-list__action"
          >
            <span>
              Count : 0
            </span>
            <button
              class="btn"
            >
              Edit
            </button>
          </div>
        </div>
        <div
          class="contact-list"
        >
          <div
            class="contact-list__desc"
          >
            <img
              alt=""
              class="profile-photo circle"
              src="https://source.unsplash.com/random/7"
            />
            <div
              class="offset-left-20"
            >
              <h3>
                 Faisal Amrullah
              </h3>
              <span>
                amrullahfaisal@gmail.com
              </span>
              <br />
              <span>
                081806868501
              </span>
            </div>
          </div>
          <div
            class="contact-list__action"
          >
            <span>
              Count : 0
            </span>
            <button
              class="btn"
            >
              Edit
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
